crescent/context.py

```diff
@@ -59,11 +111,39 @@ class Context:
     user: User
     member: Optional[Member]
 
+    command: str
+    command_type: hikari.CommandType
+    group: Optional[str]
+    sub_group: Optional[str]
+    options: Dict[str, Any]
+
     _has_replied: bool = False
     _used_first_resp: bool = False
 
     @classmethod
     def _from_command_interaction(cls: Type[Context], interaction: CommandInteraction) -> Context:
+        name: str = interaction.command_name
+        group: Optional[str] = None
+        sub_group: Optional[str] = None
+        options = interaction.options
+
+        if options:
+            option = options[0]
+            if option.type == 1:
+                group = name
+                name = option.name
+                options = option.options
+            elif option.type == 2:
+                group = interaction.command_name
+                sub_group = option.name
+                name = unwrap(option.options)[0].name
+
+        callback_options: Mapping[str, OptionTypesT | Message | User]
+        if interaction.command_type is CommandType.SLASH:
+            callback_options = _options_to_kwargs(interaction, options)
+        else:
+            callback_options = _resolved_data_to_kwargs(interaction)
```

Comment:
I don't think this belongs here. I think its better to change the method to something that has the interaction, name, groups, options, etc as parameters.